include/libtorrent/block_cache.hpp

```diff
@@ -216,7 +211,7 @@ namespace libtorrent
 		//TODO: make this 32 bits and to count seconds since the block cache was created
 		time_point expire = min_time();
 
-		std::uint64_t piece:22;
+		piece_index_t piece;
```

Comment:
well, this type cannot be made part of the bitfield, so this will widen the type